ticdc: update cdc gc service doc and failed changefeed faq (#16583)
pingcap/docs: ticdc/ticdc-faq.md

When starting the TiCDC server, you can specify the Time To Live (TTL) duration of GC safepoint by configuring `gc-ttl`. You can also [use TiUP to modify](/ticdc/deploy-ticdc.md#modify-ticdc-cluster-configurations-using-tiup) `gc-ttl`. The default value is 24 hours. In TiCDC, this value means:

- The maximum time the GC safepoint is retained at the PD after the TiCDC service is stopped.

> - When TiKV GC data is blocked by TiCDC's GC safepoint, `gc-ttl` indicates the maximum replication delay of a TiCDC replication task. If the delay of the replication task exceeds the value set by `gc-ttl`, the replication task enters into the `failed` state and reports the `ErrGCTTLExceeded` error. It cannot be recovered, and no longer blocks GC safepoint to advance.

Reviewer: In the original text "当 TiCDC 的 GC safepoint 阻塞 TiKV GC 数据时", "GC" is not very correctly used as a verb and "数据" is the object. Here it can be directly translated as "When TiKV's performing GC to data is blocked by ...", but actually the word "data" can be removed. The current translation "GC data" reads like "data about GC", which is not correct.
```suggestion
- When TiKV's GC is blocked by TiCDC's GC safepoint, `gc-ttl` indicates the maximum replication delay of a TiCDC replication task. If the delay of the replication task exceeds the value set by `gc-ttl`, the replication task enters into the `failed` state and reports the `ErrGCTTLExceeded` error. It cannot be recovered, and no longer blocks GC safepoint to advance.
```

If the replication task is suspended longer than the time specified by `gc-ttl`, the replication task enters the `failed` status and cannot be resumed. The PD corresponding service GC safepoint will continue.

> The default Time-To-Live (TTL) that TiCDC sets for a service GC safepoint is 24 hours, which means that the GC mechanism does not delete any data if the TiCDC service can be recovered within 24 hours after it is interrupted.

Author: ```suggestion
The default Time-To-Live (TTL) that TiCDC sets for a service GC safepoint is 24 hours, which means that the GC mechanism does not delete the data required by TiCDC for continuing replication if the TiCDC service can be recovered within 24 hours after it is interrupted.
```

## How to recover a replication task after it fails?

1. Use `cdc cli changefeed query` to query the error information of the replication task and fix the error as soon as possible.
2. Increase the value of `gc-ttl` to allow more time to fix the error, ensuring that the replication task does not enter the `failed` status due to the replication delay exceeding `gc-ttl` after the error is fixed.

> 3. After evaluating the impact on the system, increase the value of [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-new-in-v50) in TiDB to prevent GC from retaining data, ensuring that the replication task does not enter the `failed` status due to GC cleaning data after the error is fixed.

Reviewer: > ... to prevent GC from retaining data, ...

Retaining or deleting?

I think the original text is kind of ambiguous.

> ...以达到阻止 GC 保留数据的目的

The actual purpose is to "阻止 GC" and "保留数据" instead of "阻止 (GC 保留数据)" ...

> 3. After evaluating the impact on the system, increase the value of [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-new-in-v50) in TiDB to prevent GC from retaining data, ensuring that the replication task does not enter the `failed` status due to GC cleaning data after the error is fixed.

Author: ```suggestion
3. After evaluating the impact on the system, increase the value of [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-new-in-v50) in TiDB to block GC and retain data, ensuring that the replication task does not enter the `failed` status due to GC cleaning data after the error is fixed.
```
